plugins/delete-session-id.client.ts

```diff
@@ -0,0 +1,28 @@
+import {SESSION_KEY} from "~/utils/helpers";
+
+/**
+ * Nuxt.js plugin to delete the user's active session ID when the tab or browser is closed.
+ *
+ * This plugin uses the 'beforeunload' event to detect when the tab or browser is being closed.
+ * It checks if the session ID is still in the session storage.
+ * If it's not, it assumes the tab or browser was closed and deletes the user's active session ID.
+ * Session storage is used to avoid deleting the session ID
+ * when the user navigates to a different page within the same tab or refreshes the browser.
+ *
+ * @module delete-session-id.client
+ */
+export default defineNuxtPlugin((_nuxtApp) => {
+    const uploadSessionId = useCookie("uploadSessionId");
+
+    const deleteCookie = () => {
+        // Session storage is cleared when the tab or browser is closed.
+        // If the session ID is still in the session storage, it means the tab or browser was not closed
+        if (!sessionStorage.getItem(SESSION_KEY)) {
+            // If not, assume the tab or browser was closed and destroy the user's active session ID.
+            uploadSessionId.value = null;
+        }
+    };
+
+    // Add an event listener for the 'beforeunload' event
+    window.addEventListener("beforeunload", deleteCookie);
+});
```

Comment:
@coderabbitai that's because of the auto-imports. `useCookie` is always avaialble in the Nuxt context.

Reply:
@coderabbitai any other suggestions or observations related to this plugin or code?

Reply:
@coderabbitai would this solution resolve the "Event Listener Management" point you mentioned? This `window.removeEventListener("beforeunload", deleteCookie);` extra line was added.

```ts
import {SESSION_KEY} from "~/utils/helpers";

/**
 * Nuxt.js plugin to delete the user's active session ID when the tab or browser is closed.
 *
 * This plugin uses the 'beforeunload' event to detect when the tab or browser is being closed.
 * It checks if the session ID is still in the session storage.
 * If it's not, it assumes the tab or browser was closed and deletes the user's active session ID.
 * Session storage is used to avoid deleting the session ID
 * when the user navigates to a different page within the same tab or refreshes the browser.
 *
 * @module delete-session-id.client
 */
export default defineNuxtPlugin((_nuxtApp) => {
    const uploadSessionId = useCookie("uploadSessionId");

    const deleteCookie = () => {
        // Session storage is cleared when the tab or browser is closed.
        // If the session ID is still in the session storage, it means the tab or browser was not closed
        if (!sessionStorage.getItem(SESSION_KEY)) {
            // If not, assume the tab or browser was closed and destroy the user's active session ID.
            uploadSessionId.value = null;
            
            window.removeEventListener("beforeunload", deleteCookie);
        }
    };

    // Add an event listener for the 'beforeunload' event
    window.addEventListener("beforeunload", deleteCookie);
});
```

Reply:
@coderabbitai I've considered this, but it's not OK to use `onUnmounted()` since it generates a Vue warning:
```
[Vue warn]: onUnmounted is called when there is no active component instance to be associated with. Lifecycle injection APIs can only be used during execution of setup(). If you are using async setup(), make sure to register lifecycle hooks before the first await statement.
```